packages/flutter_test/test/animation_sheet_test.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('AnimationSheetBuilder', () {
+    testWidgets('correctly records frames', (WidgetTester tester) async {
+      final AnimationSheetBuilder builder = AnimationSheetBuilder(size: _DecuplePixels.size);
+
+      await tester.pumpFrames(
+        builder.record(
+          const _DecuplePixels(Duration(seconds: 1)),
+        ),
+        const Duration(milliseconds: 200),
+        const Duration(milliseconds: 100),
+      );
+
+      await tester.pumpFrames(
+        builder.record(
+          const _DecuplePixels(Duration(seconds: 1)),
+          recording: false,
+        ),
+        const Duration(milliseconds: 200),
+        const Duration(milliseconds: 100),
+      );
+
+      await tester.pumpFrames(
+        builder.record(
+          const _DecuplePixels(Duration(seconds: 1)),
+          recording: true,
+        ),
+        const Duration(milliseconds: 400),
+        const Duration(milliseconds: 100),
+      );
+
+      final Widget display = await builder.display();
+      await tester.binding.setSurfaceSize(builder.sheetSize());
+      await tester.pumpWidget(display);
+
+      await expectLater(find.byWidget(display), matchesGoldenFile('test.animation_sheet_builder.records.png'));
```

Comment:
This test may need to be moved to `package:flutter` so it will go to the Flutter Gold dashboard. Calls to `matchesGoldenFile` in `package:flutter_test` don't use the FlutterGoldenFileComparator.